infra/role.tf

```diff
@@ -13,13 +13,28 @@ resource "aws_iam_role" "app_role" {
       },
     ]
   })
-}
+  inline_policy {
+    name = "allow_s3_access"
 
-resource "aws_iam_role_policy_attachment" "ecs_execution" {
-  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
-  role       = aws_iam_role.app_role.name
+    policy = jsonencode({
```

Comment:
(just for my own understanding) what's the upshot of pulling this out of the `resource "aws_iam_role_policy" "s3"{ ...` block into its own thing?

Reply:
No real rhyme or reason on which syntax I'm going with other than the example documentation I happen to be looking at and got working at the time of writing. The motivation to change this was that I had the policy on the wrong role. (It was set up so the task _execution_ role was able to manipulate S3 objects but we actually needed the _task_ itself to talk to S3.) And then ... this is just the particular way I got it working.